### PR TITLE
Add missing exception handeling

### PR DESCRIPTION
**Pull request by Flova**
The case that a message is received before the first camera info, is only handled for message types that use `project_point` and not for ones that use `project_points`. This pull request adds this functionality.

### CODE REVIEW

**Review by ijnek — APPROVED** [review]
Looks good, did you want to handle the NoIntersectionError too while we're at it?

**Comment by Flova** [comment]
> Looks good, did you want to handle the NoIntersectionError too while we're at it?

The no intersection error is not present for the vectorized case as it is common for some elements of the vector to be not projectable. For example a few pixels in an image above the horizon. Throwing an exception would abort the projection of all elements. Therefore the elements which are not projectable are set to Nan and handled separately. The no intersection error is catched in all cases where a single projection per IPM call is used.

**Comment by ijnek** [comment]
Got it, happy for you to merge!

File: soccer_ipm/soccer_ipm/msgs/field_boundary.py [hunk 1]
```diff
@@ -46,12 +47,16 @@ def map_field_boundary(
...
+    try:
+        points_on_plane = ipm.map_points(
+            field,
+            points_np,
+            msg.header.stamp,
+            plane_frame_id=output_frame,
+            output_frame_id=output_frame)[1]
+    except CameraInfoNotSetException:
+        logger.warn('Inverse perspective mapping should be performed, \
+                but no camera info was recived yet!', throttle_duration_sec=5)
```
Comment (jaagut) [inline soccer_ipm/soccer_ipm/msgs/field_boundary.py]: I am not certain, how this gets shown by the logger, but breaking up a string like this in normal python leads to a lot of spaces before `but...`.
Rather, use `+` for concatenation.

File: soccer_ipm/soccer_ipm/msgs/markings.py [hunk 2]
```diff
@@ -157,12 +157,16 @@ def map_marking_segments(
...
+    try:
+        segments_on_plane = ipm.map_points(
+            field,
+            segment_points_np.reshape(-1, 2),
+            header.stamp,
+            plane_frame_id=output_frame,
+            output_frame_id=output_frame)[1].reshape(-1, 2, 3)
+    except CameraInfoNotSetException:
+        logger.warn('Inverse perspective mapping should be performed, \
+                but no camera info was recived yet!', throttle_duration_sec=5)
```
Comment (jaagut) [inline soccer_ipm/soccer_ipm/msgs/markings.py]: Same as above

**Comment by Flova** [comment]
I needed to rework this a bit. In the current state the exception was handled, but we still published an empty array of e.g. balls or the code would crash in the case of the point clouds. I therefore put the exception handling a layer higher, above the publishing call. This is done using a python decorator annotation.

File: soccer_ipm/soccer_ipm/utils.py [hunk 3]
```diff
@@ -56,3 +61,26 @@ def bb_footpoint(bounding_box: BoundingBox2D) -> Point2D:
         x=float(bounding_box.center.position.x),
         y=float(bounding_box.center.position.y + bounding_box.size_y // 2),
     )
+
+
+def catch_camera_info_error(logger: RcutilsLogger) -> Callable:
```
Comment (jaagut) [inline soccer_ipm/soccer_ipm/utils.py]: This method name is not precise.
I'd prefer `catch_camera_info_not_set_error`

**Review by ijnek — APPROVED** [review]

**Pull request merged**